File: php/src/Creational/AbstractFactory/BostonConcreteSoupFactory.php
<?

#//BostonConcreteSoupFactory.java - One of Two concrete factories extending the abstract factory

require_once 'AbstractSoupFactory.php';

class BostonConcreteSoupFactory extends AbstractSoupFactory
{
	function __construct()
	{
			$this->factoryLocation = "Boston";
	}

	function makeClamChowder() 
	{
		return new BostonClamChowder();
	}

	function makeFishChowder() 
	{
		return new BostonFishChowder();
	}
}

File: go/creational/abstractfactory/BostonConcreteSoupFactory.go
//BostonConcreteSoupFactory - One of Two concrete factories extending the abstract factory

package abstractfactory

import "../common"

type BostonConcreteSoupFactory struct {
	BaseSoupFactory
}

func NewBostonConcreteSoupFactory () *BostonConcreteSoupFactory {
	return &BostonConcreteSoupFactory{ BaseSoupFactory{"Boston"}}
}

func (this *BostonConcreteSoupFactory) MakeClamChowder() common.Soup {
	return common.NewBostonClamChowder()
}

func (this *BostonConcreteSoupFactory) MakeFishChowder() common.Soup {
	return common.NewBostonFishChowder()
}

